Add deauth tile

Nels885 opened the pull request:
Added the Deauth tile following the issue #41 

I put a video on the Discord where you can see how this new tile works on my T-Watch v2.

dglaude commented:
Hi, I just compiled with 4.4.2 and tested on v3 Watch.
* Graphically it is great and like the "Rogue AP" with the graphical glitch #39 so when that one is fixed, the fix need to be shared.
* Practically, I tested Deauth and it works super great (maybe there should be an enhanced version where you can deauth a specific host on a Wifi and not the whole wifi... well I assume it is possible, would require a scan of the active host on a specific SSID).
* Functionnaly, I had a Freeze in the new Deauth menu, and I think there was no Deauth active. I was on that page, I tested multiple time Start and Stop. Then I left it a little bit doing nothing on that page... but then nothing was responding and I could not start it or escape from it.
* Codewise... I did not look at the code, I just hope there is some re-use for the AP selection part, but maybe that would make the compilation more complex, you would need that piece of code if one of the two Tile that use it is activated... maybe duplication of the code is easier in the current use case ???

Maybe the main scan need to be modified to make it "read-only" because now there are two locations to do Deauth (but that is another change to another Tile.

It's really great somebody took the idea and implemented it!
So fast!!!

Nels885 commented:
For the moment, I only ported the code of the RogueAP tile and I modified it to integrate the Deauth to avoid too big a modification of the code and the risk of adding new bugs.
I didn't make any correction about the graphical glitch, but if it is fixed, it will also be applied on the new tile.
The select AP part is not reused, it is initialized in each of the tiles.

There are many improvements to review but this requires major code changes that cannot be applied for version 2.0 which should be available Tuesday evening:

- The scanner is permanently active on the Deauth and RogueAP tiles, I think it would be better to integrate the scanner in the Select AP modal but that requires revising the design of the modals or creating a specific modal.
- Added the ability to select a specific client, this part does not exist at the moment.
- Remove the select AP part of the tiles to move it into the wifi module.

virtualabs commented:
Graphical issues will be fixed next Tuesday as the bugs lie in my *twatch-lib* library and not in this repository. These issues should be quite simple to fix and I hope it won't take too much time. As @Nels885 wrote, the main goal for now is to reach a working version of HackWatch version 2 including WiFi and BLE features, we will take care of the proposed enhancements in version 2.1 I think. 

However, this freeze you experienced while trying the deauth feature is still concerning. I'll try to reproduce and fix this before releasing v2.
